data-runtime/src/main/java/io/micronaut/data/runtime/intercept/criteria/AbstractSpecificationInterceptor.java

```diff
@@ -150,7 +152,9 @@ protected final Iterable<?> findAll(RepositoryMethodKey methodKey, MethodInvocat
             }
             return criteriaRepositoryOperations.findAll(query);
         }
-        return operations.findAll(preparedQueryForCriteria(methodKey, context, type, methodJoinPaths));
+        PreparedQuery<?, ?> preparedQuery = preparedQueryForCriteria(methodKey, context, type, methodJoinPaths);
+        context.setAttribute(PREPARED_QUERY_KEY, preparedQuery);
```

Comment:
Don't know if this is safe to be used, tests say it is ok. Since cursored page processing is done in the prepared query, wanted to reuse it this way.